spec/helpers/time_options_helper_spec.rb

```diff
@@ -0,0 +1,31 @@
+require "rails_helper"
+
+RSpec.describe TimeOptionsHelper do
+  describe "#time_options" do
+    subject { helper.time_options }
+
+    it "contains a 12:01am special case" do
+      expect(subject).to include "12:01am"
+    end
+
+    it "does not contain 12:00am special case" do
+      expect(subject).not_to include "12:00am"
+    end
+
+    it "returns 48 items (hourly - except 12am special case - and half-hourly)" do
+      expect(subject.count).to eq(48)
+    end
+
+    it "returns all the AM times before the PM times" do
+      index_of_last_am = subject.each_with_index.map { |time, index| time.end_with?("am") ? index : nil  }.compact.last
```

Comment:
You could use a [select](https://www.shortcutfoo.com/app/dojos/ruby-arrays/cheatsheet) here to just get all of the "am" values.

Reply:
That's not quite what this is doing though - I'm getting the position of the last "am" value in the array so that I can make sure it comes before the first "pm" value in the array, thereby ensuring all "am" values are before all "pm" values.

A select would filter the "am"/"pm" array into just an array of "am" values and lose the index information I need - unless I'm misunderstanding your suggestion :)

Reply:
I assumed you could "select" all of the "am"s and then do a count-1 to get the index of the last one.

Reply:
Unfortunately that wouldn't work, imagine:

```ruby
mixed = ["1am", "2am", "9pm", "3am", "7pm"]

index_of_last_am = mixed.select { |time| time.end_with?("am") }.count - 1
# ^ would become ["1am", "2am", "3am"], therefore index = 3-1 = 2
# when actually at index 2 we have "9pm"
```